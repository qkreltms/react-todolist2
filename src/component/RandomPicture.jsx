import React from "react";
import * as actions from "../actions/randomDogs";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { FormattedMessage } from "react-intl";

const RandomPicture = props => {
  return (
    <div>
      <button onClick={() => props.dispatch(actions.fetchDog())}>
        <FormattedMessage id="show_dog" />
      </button>
      {props.loading ? (
        <p><FormattedMessage id="loading" /></p>
      ) : props.error ? (
        <p>{props.errorMsg}</p>
      ) : (
        <p>
          <img alt="img for dogs" src={props.url} />
        </p>
      )}
    </div>
  );
};

RandomPicture.propTypes = {
  url: PropTypes.string,
  error: PropTypes.bool,
  errorMsg: PropTypes.string,
  loading: PropTypes.bool,
  dispatch: PropTypes.func
};

RandomPicture.deafultProps = {
  url: "",
  error: false,
  errorMsg: "",
  lading: false,
  dispatch: null
};

const mapStateToProps = state => ({
  url: state.randomDogs.url,
  error: state.randomDogs.error,
  errorMsg: state.randomDogs.errorMsg,
  loading: state.randomDogs.loading
});

export default connect(mapStateToProps)(RandomPicture);
