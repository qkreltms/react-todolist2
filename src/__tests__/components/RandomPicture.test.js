import React from 'react';
import RandomPicture from '../../component/RandomPicture';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('RandomPicture', () => {
    test('renders without crashing', () => {
        const component = shallow(<RandomPicture />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
      })
})
