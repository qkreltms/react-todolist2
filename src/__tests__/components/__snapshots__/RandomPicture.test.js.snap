// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomPicture renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
